### ISSUE TITLE
"The model: `gpt-4` does not exist error" on --gpt4only flag

### ISSUE THREAD

**Issue by PianistaMichal** [Significant-Gravitas/Auto-GPT#1856]
### Duplicates

- [X] I have searched the existing issues

### Steps to reproduce 🕹

Run `python -m autogpt --gpt4only` fill AI name, purpose and goals. Then after code will work by itself error is thrown

### Current behavior 😯

This error is thrown:

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/appuser/autogpt/__main__.py", line 572, in <module>
    main()
  File "/home/appuser/autogpt/__main__.py", line 396, in main
    agent.start_interaction_loop()
  File "/home/appuser/autogpt/__main__.py", line 448, in start_interaction_loop
    assistant_reply = chat.chat_with_ai(
                      ^^^^^^^^^^^^^^^^^^
  File "/home/appuser/autogpt/chat.py", line 159, in chat_with_ai
    assistant_reply = create_chat_completion(
                      ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/appuser/autogpt/llm_utils.py", line 39, in create_chat_completion
    response = openai.ChatCompletion.create(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/appuser/.local/lib/python3.11/site-packages/openai/api_resources/chat_completion.py", line 25, in create
    return super().create(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/appuser/.local/lib/python3.11/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 153, in create
    response, _, api_key = requestor.request(
                           ^^^^^^^^^^^^^^^^^^
  File "/home/appuser/.local/lib/python3.11/site-packages/openai/api_requestor.py", line 226, in request
    resp, got_stream = self._interpret_response(result, stream)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/appuser/.local/lib/python3.11/site-packages/openai/api_requestor.py", line 619, in _interpret_response
    self._interpret_response_line(
  File "/home/appuser/.local/lib/python3.11/site-packages/openai/api_requestor.py", line 682, in _interpret_response_line
    raise self.handle_error_response(
openai.error.InvalidRequestError: The model: `gpt-4` does not exist
```

### Expected behavior 🤔

Code should work without issues on gpt4

### Your prompt 📝

```yaml
any prompt fails
```

**Comment by hemangjoshi37a** [comment]
Solution
It seems like the gpt-4 model is not available in the OpenAI API. This could be because it hasn't been released yet or it could be a typo. To fix the error, the user could try using a different model, such as gpt-3, or double-check the spelling of the model name.

Additionally, the user could check if they have the latest version of the autogpt package installed and update it if necessary. They could also try reinstalling the package and its dependencies to ensure that everything is up to date and installed correctly.

If the issue persists, the user could try reaching out to the maintainers of the autogpt package for further assistance or submit a bug report.

**Comment by JoeKanaan0** [comment]
Why does the comment above me sound like It was written by AI?? Anyways gpt-4 isn't available to everyone  you need to join the openAI waitlist and you can get it faster if you are a developer.

**Comment by Jefferydo** [comment]
I'm getting the error routinely. It's single-handedly destroying every query I attempt. Trying to stay interested, but I have yet to get this app to do anything other than churn away, find promising resources and fail to act on them.

I purchased a paying account here, presumably unlocking GPT-4: https://platform.openai.com/account/billing/overview

**Comment by Bigyeti-Tech** [comment]
I have also tried this python -m autogpt --gpt4only
It starts, and I get
GPT4 Only Mode:  ENABLED
Welcome back!  Would you like me to return to being FooBar-GPT?
Goals: ['Some goals']
Continue (y/n): y
Using memory of type: PineconeMemory

Then it crashes.

**Comment by oojBuffalo** [comment]
I also have a sub to ChatGPT Plus as well as access to GPT-4 API, however I always get the error above when the agent tries to use it.  I did not use the --gpt4only flag though - the agent mostly uses GPT-3.5 and occasionally switches to GPT-4 (or at least attempts to) in order to write/run code.

**Comment by RoySRose** [comment]
>After reading carefully, I don't believe direct GPT-4 API access is included with ChatGPT Plus.

this I didn't know. I thought GPT-4 API will of course be included with ChatGPT Plus. 
@oojBuffalo any links that you can share?

**Comment by psvensson** [comment]
It seems that there is some kind of problem on the openAI side. More people have gpt-4 API issues; https://community.openai.com/t/i-cant-use-the-gpt-4-version-i-am-paying/169061/3

I have also tried a command-line tool to test if the problem persists - and it does!
https://github.com/CristiVlad25/gpt-cli

So the problem is on the openAI side.

Noet: I have been paying for the OpenAI API and used gpt-4 with much problems the last three week. It stoppe dead in the water Thursday the 20/4.

**Comment by ntindle** [comment]
To the best of my knowledge this is an open ai bug not ours. Please reopen if I’m incorrecr

**Comment by Delta-Ark** [comment]
How can we track when this problem is fixed? Just try it randomly and wait for them to fix the bug? (I have it too).

**Comment by missioncfwf** [comment]
The fix is to use 3.5-turbo instead of 4, but does anyone know which file that change js in?

**Comment by Delta-Ark** [comment]
I understand; OK, I applied for access.
